src/ray/common/id.cc

```diff
@@ -104,23 +173,60 @@ TaskID ObjectID::TaskId() const {
       std::string(reinterpret_cast<const char *>(id_), TaskID::Size()));
 }
 
-ObjectID ObjectID::ForPut(const TaskID &task_id, int64_t put_index) {
-  RAY_CHECK(put_index >= 1 && put_index <= kMaxTaskPuts) << "index=" << put_index;
+ObjectID ObjectID::ForPut(const TaskID &task_id, uint32_t put_index) {
```

Comment:
To reduce code duplication and potentially future errors, this, `ObjectID::ForTaskReturn`, and potentially `ObjectID::FromRandom` should share a common private method to generate the `ObjectID`.